pkg/gardenlet/controller/managedseed/managedseed_controller_test.go

```diff
@@ -106,24 +134,45 @@ var _ = Describe("Controller", func() {
 
 			c.managedSeedAdd(managedSeed)
 		})
+
 	})
 
 	Describe("#managedSeedUpdate", func() {
 		It("should do nothing because the new object is not a ManagedSeed", func() {
 			c.managedSeedUpdate(nil, &gardencorev1beta1.Seed{})
 		})
 
+		It("should add the object to the queue", func() {
+			queue.EXPECT().Add(key)
+
+			c.managedSeedUpdate(nil, managedSeed)
+		})
+
 		It("should do nothing because the object generation and observed generation are equal", func() {
-			managedSeed.Status.ObservedGeneration = 1
+			managedSeed.Status.ObservedGeneration = 2
 
 			c.managedSeedUpdate(nil, managedSeed)
 		})
 
-		It("should add the object to the queue", func() {
-			queue.EXPECT().Add(key)
+		It("should add the object to the queue with a jittered delay because object generation and observed generation are not equal and jitterUpdates is set to true", func() {
+			managedSeed.Status.ObservedGeneration = 3
+			c.config.Controllers.ManagedSeed.JitterUpdates = pointer.Bool(true)
+			queue.EXPECT().AddAfter(key, gomock.AssignableToTypeOf(time.Second)).DoAndReturn(
+				func(_ interface{}, d time.Duration) {
+					Expect(d > 0 && d <= syncJitterPeriod).To(BeTrue())
+				},
+			)
```

Comment:
does it work to use a helper function for this like expectAddedWithJitter() as this code is duplicated a lot